starter-code/src/contexts/CountryContext.js

```diff
@@ -0,0 +1,21 @@
+import React, {createContext, useState} from 'react';
+import countriesJson from './../countries.json';
+
+export const CountryContext = createContext();
+
+const CountryContextProvider = (props) => {
+  const [countries, setCountries] = useState(countriesJson);
+
+  const findCountry = (cca3) => {
+    return countries.find(country => country.cca3 === cca3);
+  }
+
+  return(
+    <CountryContext.Provider value={{countries, setCountries, findCountry}}>
+      {props.children}
+    </CountryContext.Provider>
+  );
+}
+
+export default CountryContextProvider;
+
```

Comment:
Chicos el haber utilizado `Context` es 🔝. Se está trabajando mucho de esta manera en la industria y si lo cogéis desde el principio os va a ayudar mucho durante el último proyecto.

Reply:
Intentaremos hacer más uso del destructuring a partir de ahora porque es verdad que resulta más cómodo. Muchas gracias por tus comentarios, Diego!